stock_release_channel/models/stock_release_channel.py

```diff
@@ -472,6 +472,10 @@ def assign_release_channel(self, picking):
         ):
             return
         message = ""
+        partner_release_channel = picking.partner_id._get_partner_release_channel()
```

Comment:
IMO we should add a m2m also on the release channel and display this field on the channel form. Moreover, we must add something in place to instruct the user that when a partner is assigned to a channel, others criteria are no more evaluated. I've the feeling this rule functionality should be implemented into a dedicated addon since it break the normal flow selection flow.

Reply:
ping @jbaudoux